Alignment/OfflineValidation/plugins/SagittaBiasNtuplizer.cc

```diff
@@ -0,0 +1,550 @@
+  desc.add<double>("muondxySigCut", 4.)->setComment("significance of the d0 compatibility with closest vertex");
+  desc.add<double>("minMassWindowCut", 70.)->setComment("in GeV");
+  desc.add<double>("maxMassWindowCut", 110.)->setComment("in GeV");
+  desc.add<double>("d0CompatibilityCut", 0.01)->setComment("d0 compatibility between the two muons");
+  desc.add<double>("z0CompatibilityCut", 0.06)->setComment("z0 compatibility between the two muons");
+  desc.add<std::vector<double>>("pTThresholds", {30., 10.});
+  descriptions.addWithDefaultLabel(desc);
+}
+
+//_________________________________________________________________________________
+void SagittaBiasNtuplizer::analyze(const edm::Event& event, const edm::EventSetup& setup) {
+  h_cutFlow->Fill(0);
+
+  // the di-muon tracks
+  std::vector<const reco::Track*> myTracks;
+
+  // if we have to start from scratch from RECO data-tier
+  if (useReco_) {
+    // select the good muons
+    std::vector<const reco::Muon*> myGoodMuonVector;
+    for (const auto& muon : event.get(muonsToken_)) {
+      const reco::TrackRef t = muon.innerTrack();
+      if (!t.isNull()) {
+        if (t->quality(reco::TrackBase::highPurity)) {
+          if (t->chi2() / t->ndof() <= 2.5 && t->numberOfValidHits() >= 5 &&
+              t->hitPattern().numberOfValidPixelHits() >= 2 && t->quality(reco::TrackBase::highPurity))
+            myGoodMuonVector.emplace_back(&muon);
+        }
+      }
+    }
+
+    LogDebug("SagittaBiasNtuplizer") << "myGoodMuonVector size: " << myGoodMuonVector.size() << std::endl;
+    std::sort(myGoodMuonVector.begin(), myGoodMuonVector.end(), [](const reco::Muon*& lhs, const reco::Muon*& rhs) {
+      return lhs->pt() > rhs->pt();
+    });
+
+    // just check the ordering
+    for (const auto& muon : myGoodMuonVector) {
+      LogDebug("SagittaBiasNtuplizer") << "pT: " << muon->pt() << " ";
+    }
+    LogDebug("SagittaBiasNtuplizer") << std::endl;
+
+    // reject if there's no Z
+    if (myGoodMuonVector.size() < 2)
+      return;
+
+    if ((myGoodMuonVector[0]->pt()) < pTthresholds_[0] || (myGoodMuonVector[1]->pt() < pTthresholds_[1]))
+      return;
+
+    if (myGoodMuonVector[0]->charge() * myGoodMuonVector[1]->charge() > 0)
+      return;
+
+    //const auto& m1 = myGoodMuonVector[1]->p4();
+    //const auto& m0 = myGoodMuonVector[0]->p4();
+    //const auto& mother = m0 + m1;
+
+    // just copy the top two muons
+    std::vector<const reco::Muon*> theZMuonVector;
+    theZMuonVector.reserve(2);
+    theZMuonVector.emplace_back(myGoodMuonVector[1]);
+    theZMuonVector.emplace_back(myGoodMuonVector[0]);
+
+    // do the matching of Z muons with inner tracks
+    unsigned int i = 0;
+    for (const auto& muon : theZMuonVector) {
+      i++;
+      float minD = 1000.;
+      const reco::Track* theMatch = nullptr;
+      for (const auto& track : event.get(tracksToken_)) {
+        float D = ::deltaR(muon->eta(), muon->phi(), track.eta(), track.phi());
```

Comment:
`deltaR2` is preferable in terms of performance

Reply:
OK, I don't think performance here is of paramount importance, but in any case it was applied in the last push.